conf/bkenv.sh

```diff
@@ -33,10 +33,13 @@ BOOKIE_CONF=${BOOKIE_CONF:-"$BK_HOME/conf/bookkeeper.conf"}
 # BOOKIE_LOG_DIR=
 
 # Memory size options
-BOOKIE_MEM=" -Xms2g -Xmx2g -XX:MaxDirectMemorySize=2g"
+BOOKIE_MEM=" -Xmx2g -XX:MaxDirectMemorySize=2g -XX:+ExitOnOutOfMemoryError"
 
 # Garbage collection options
-BOOKIE_GC=" -XX:+UseG1GC -XX:MaxGCPauseMillis=10 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+AggressiveOpts -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC -XX:-ResizePLAB"
+BOOKIE_GC=" -XX:+UseG1GC"
```

Comment:
This will be a breaking change for people who use default settings to run pulsar at production.

I would suggest keeping the default but allow people to override it through system environment.

```
BOOKIE_GC=${BOOKIE_GC:-" -XX:+UseG1GC -XX:MaxGCPauseMillis=10 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+AggressiveOpts -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC -XX:-ResizePLAB"}
```

Reply:
I don't think this is a "breaking" change in the sense that it will not make anything that was work to stop working. 

Sure, it's a different setting and that should be documented, but the purpose of this change is to change the "default" setting, because it's not ideal to be used as "default", rather than allow the users to change the default (which is already possible).

Reply:
To give context, these flags were the settings we were using in a very particular hardware configuration in production. They work well in prod/performance scenarios in that configuration but they probably don't work that well in any other slightly different configuration. 

Default should indeed be more conservative and work reasonably well under most conditions. It might not be super-tuned for low-latency or high throughput, and we can give some guidelines in documentation around that, but if one cares about some performance aspect, he would have to test anyway in its own environment anyway.

Reply:
I understand the motivation. However we should also be taking upgrading existing production users into account. Otherwise, it can be a huge surprise to the users. I treat such surprises as "breaking changes". 

I would prefer a documentation and an entry in release notes for this, before merging this change.

Reply:
Is bookie GC really an issue? Doesn't the ack quorum being smaller than the write quorum mask most GC blips?